dgidb/network_graph.py

```diff
@@ -0,0 +1,115 @@
+import networkx as nx
+import plotly.graph_objects as go
+
+def initalize_network(interactions):
+    interactions_graph = nx.Graph()
+    for int_gene, int_drug in zip(interactions['gene'], interactions['drug']):
+        interactions_graph.add_node(int_gene,isGene=True)
+        interactions_graph.add_node(int_drug,isGene=False)
+        interactions_graph.add_edge(int_gene,int_drug)
+    return interactions_graph
+
+def add_node_attributes(interactions_graph):
+    for node in interactions_graph.nodes:
+        is_gene = interactions_graph.nodes[node]['isGene']
+        if(is_gene):
+            set_color = "cyan"
+            set_size = 10
+        else:
+            degree = interactions_graph.degree[node]
+            if(degree > 1):
+                set_color = "orange"
+                set_size = 7
+            else: 
+                set_color = "red"
+                set_size = 7
+        interactions_graph.nodes[node]['node_color'] = set_color
+        interactions_graph.nodes[node]['node_size'] = set_size
+
+def create_network(interactions):
+    interactions_graph = initalize_network(interactions)
+    add_node_attributes(interactions_graph)
+    return interactions_graph
+
+def generate_plotly(graph):
+    layout = go.Layout(
+        hovermode='closest',
+        xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+        yaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+        showlegend=True
+    )
+    fig = go.Figure(layout=layout)
+    
+    if(graph is not None):
+        pos = nx.spring_layout(graph, seed=7)
+
+        trace_nodes = create_trace_nodes(graph,pos)
+        trace_edges = create_trace_edges(graph,pos)
+
+        fig.add_trace(trace_edges)
+        for trace_group in trace_nodes:
+            fig.add_trace(trace_group)
+
+    return fig
+
+def create_trace_nodes(graph,pos):
+    nodes_by_group = {
+        'cyan': {'node_x': [], 'node_y': [], 'node_text': [], 'node_color': [], 'node_size': [], 'legend_name': "genes"},
+        'orange': {'node_x': [], 'node_y': [], 'node_text': [], 'node_color': [], 'node_size': [], 'legend_name': "multi-degree drugs"},
+        'red': {'node_x': [], 'node_y': [], 'node_text': [], 'node_color': [], 'node_size': [], 'legend_name': "single-degree drugs"}
+    }

```

Comment:
I still think we should look into seeing if there is a way to implement an easy visual indicator of approval status. I think this type of graph could be useful to identify 'promiscious', approved drugs at a glance given a cluster of genes.

Reply:
additionally, the legend is still displaying the color labels as just their respective color names. We should change this now before we move forward
![Screenshot 2023-08-23 at 12 18 20 PM](https://github.com/mcannon068nw/python-dgidb/assets/87494086/b3a3d36b-f8cc-4771-b1f8-f52267d7ed3d)